lang/csharp/src/apache/main/AvroDecimal.cs

```diff
@@ -0,0 +1,929 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Globalization;
+using System.Numerics;
+
+namespace Avro
+{
+    /// <summary>
+    /// Represents a big decimal.
+    /// </summary>
+    #pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+    public struct AvroDecimal : IConvertible, IFormattable, IComparable, IComparable<AvroDecimal>, IEquatable<AvroDecimal>
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvroDecimal"/> class from a given double.
+        /// </summary>
+        /// <param name="value">The double value.</param>
+        public AvroDecimal(double value)
+            : this((decimal)value)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvroDecimal"/> class from a given float.
+        /// </summary>
+        /// <param name="value">The float value.</param>
+        public AvroDecimal(float value)
+            : this((decimal)value)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvroDecimal"/> class from a given decimal.
+        /// </summary>
+        /// <param name="value">The decimal value.</param>
+        public AvroDecimal(decimal value)
+        {
+            var bytes = GetBytesFromDecimal(value);
+
+            var unscaledValueBytes = new byte[13];
+            Array.Copy(bytes, unscaledValueBytes, unscaledValueBytes.Length - 1);
```

Comment:
here you seem to be making `unscaledValueBytes` one byte longer than it needs to be, then relying on the fact that arrays in C# are guaranteed to have elements to be initialized to 0 to make this correct. that seems a bit obscure.. why not just use a 12 byte array?